src/server/serverUtils/contentLoader.ts

```diff
@@ -293,7 +293,7 @@ export class ContentLoader extends Disposable {
 		let contentLength = 0;
 
 		while (i < workspaceDocuments.length) {
-			if (PathUtil.PathEquals(readPath, workspaceDocuments[i].fileName)) {
+			if (PathUtil.PathEquals(readPath, workspaceDocuments[i].fileName) || (workspaceDocuments[i].uri.scheme === 'vscode-chat-code-block' && workspaceDocuments[i].uri.with({ fragment: '' }).toString() === readPath)) {

```

Comment:
Could put this thing `.uri.with({ fragment: '' }).toString()` in some utility, where's a good spot?